### PR TITLE
CHK-611: Update to use "@vtex/order-items" package

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

This PR updates this app to use the new `@vtex/order-items` package.

#### How to test it?

[Workspace](https://newcontextual--checkoutio.myvtex.com/)

You can use the above workspace, which currently has this package linked and also `order-manager` (vtex-apps/order-manager#51).

#### Screenshots or example usage:

N/A

#### Describe alternatives you've considered, if any.

N/A

#### Related to / Depends on

Depends on vtex/order-items#13

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -35,10 +36,11 @@
...
-    "vtex.checkout-resources": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.checkout-resources@0.45.0/public/@types/vtex.checkout-resources",
+    "vtex.checkout-graphql": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.checkout-graphql@0.60.0/public/@types/vtex.checkout-graphql",
+    "vtex.checkout-resources": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.checkout-resources@0.46.0/public/@types/vtex.checkout-resources",
     "vtex.checkout-splunk": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.checkout-splunk@0.1.0/public/@types/vtex.checkout-splunk",
-    "vtex.order-manager": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.order-manager@0.9.0/public/@types/vtex.order-manager",
-    "vtex.render-runtime": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.render-runtime@8.128.2/public/@types/vtex.render-runtime"
+    "vtex.order-items": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.order-items@0.12.0/public/@types/vtex.order-items",
+    "vtex.order-manager": "https://newcontextual--checkoutio.myvtex.com/_v/private/typings/linked/v1/vtex.order-manager@0.10.0+build1619527670/public/@types/vtex.order-manager",
```
Comment (silviasfon) [inline react/package.json:43]: I think this has to be changed for the build to work

Comment (lucasecdb) [reply react/package.json:43]: it would actually fail for another reason, I'm waiting for vtex/order-items#13 before merging

**Review by silviasfon — APPROVED** [review]

File: react/typings/globals.d.ts [hunk 2]
```diff
@@ -41,7 +43,7 @@ declare global {
   }
 
   interface MarketingData {
-    coupon: string
+    coupon: Maybe<string>
```
Comment (wdsrocha) [inline react/typings/globals.d.ts:46]: it should be optional, no?
```suggestion
    coupon?: Maybe<string>
```

**Review by wdsrocha — COMMENTED** [review]
Could you update the README as well? I would update like:
- briefly explain and link to vtex/order-items
- remove API explanation (we should open a PR updating the docs on the other repo, of course)

File: react/typings/globals.d.ts (hunk 2 above)
Comment (lucasecdb) [reply react/typings/globals.d.ts:46]: it shouldn't matter much in this case, the issue lies when you want to cast an optional field into a required one, but it is fine the other way around

**Comment by lucasecdb** [comment]
@wdsrocha considering that the other repository is private, would you still want to remove the docs from this one?

**Comment by wdsrocha** [comment]
good point, but why we don't make the other repository public? I _guess_ that it is private just because we forgot to update it

**Comment by lucasecdb** [comment]
I don't know why it is private either 😅, I'm okay with opening it since it is pratically the same as this one

**Comment by wdsrocha** [comment]
so ok with moving the API explanation there?

**Comment by lucasecdb** [comment]
sure!

**Review by wdsrocha — APPROVED** [review]
Good job!!

**Review by wdsrocha — APPROVED** [review]

**Comment by lucasecdb** [comment]
hello @vtex-apps/store-framework-devs @vtex-apps/technical-writers, can you review this pr?

**Pull request merged**